iterable-ingress-extension/src/main/java/com/mparticle/ext/iterable/IngressQueueManager.java

```diff
@@ -17,12 +24,33 @@ public IngressQueueManager(SqsClient sqsClient, String queueUrl) {
 
   public void enqueueMessage(String message) {
     SendMessageRequest req =
-        SendMessageRequest.builder().queueUrl(queueUrl).messageBody(message).build();
-    sqsClient.sendMessage(req);
+            SendMessageRequest.builder().queueUrl(queueUrl).messageBody(message).build();
+
+    int attempt = 0;
+    while (true) {
+      attempt++;
+      try {
+        sqsClient.sendMessage(req);
+        break;
+      } catch (SdkClientException e) {
+        StringWriter sw = new StringWriter();
+        e.printStackTrace(new PrintWriter(sw));
+        System.out.println("Failed attempt " + attempt + " to upload message to SQS  \n" + sw.toString());
+        if (attempt == NUM_RETRIES) {
+          throw e;
+        }
+      }
+    }
   }
 
   public static IngressQueueManager create() {
-    SqsClient sqsClient = SqsClient.builder().region(Region.US_EAST_1).build();
+    ClientOverrideConfiguration clientConfig = ClientOverrideConfiguration.builder()
+            .apiCallAttemptTimeout(Duration.ofSeconds(CLIENT_TIMEOUT_SECONDS))
```

Comment:
Check the default retry policy of the AWS client. It handles both the timeout and retries. If it handles retries, we may not need extra code on our side for that.
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.Builder.html#retryPolicy-software.amazon.awssdk.core.retry.RetryPolicy-

Reply:
@vbabenkoru Good call. I wrote up my findings on retries in the description ^^

I couldn't find the default timeout, but I don't see any harm in explicitly setting it.